### PR TITLE
fix(issue-510): support Additional state to be passed as the IntersectionObserverEntry

### PR DESCRIPTION
**Pull request by xinlli**
- support Additional state to be passed as the IntersectionObserverEntry
- yarn test

### CODE REVIEW

File: addon-test-support/did-intersect-mock.js [hunk 1]
```diff
@@ -39,22 +39,26 @@ class MockIntersectionObserver {
   /**
    * Force a single element to enter the viewport
    * @param {String} el - a DOM selector string
+   * @param {object} [state] Additional state to be passed as the IntersectionObserverEntry
```
Comment (elwayman02) [inline addon-test-support/did-intersect-mock.js:42]: Lets type this as `IntersectionObserverEntry` instead of `object`?

**Comment by elwayman02** [comment]
Can you update the testing docs as well to note people can construct and pass their own IntersectionObserverEntry?

https://github.com/elwayman02/ember-scroll-modifiers/blob/master/docs/modifiers/did-intersect.md#testing

**Comment by xinlli** [comment]
> Can you update the testing docs as well to note people can construct and pass their own IntersectionObserverEntry?
> 
> https://github.com/elwayman02/ember-scroll-modifiers/blob/master/docs/modifiers/did-intersect.md#testing

fixed

File: addon-test-support/did-intersect-mock.js (hunk 1 above)
Comment (xinlli) [reply addon-test-support/did-intersect-mock.js]: fixed

**Review by elwayman02 — APPROVED** [review]

**Pull request merged**